solr/core/src/java/org/apache/solr/servlet/SolrDispatchFilter.java

```diff
@@ -456,6 +457,7 @@ public void doFilter(ServletRequest _request, ServletResponse _response, FilterC
 
       GlobalTracer.get().clearContext();
       consumeInputFully(request, response);
+      SolrRequestInfo.reset();
```

Comment:
there we need to check that stack have fully cleared and disposed

Reply:
That is what `reset` _does_